daq/dhcp_monitor.py

```diff
@@ -68,9 +68,9 @@ def cleanup(self):
             self.dhcp_log.close()
             self.dhcp_log = None
         if self.dhcp_traffic:
-            if self.dhcp_traffic.stream() and not self.dhcp_traffic.stream().closed:
+            if self.dhcp_traffic.stream():
```

Comment:
self.dhcp_traffic.stream() will be truthy until self.dhcp_traffic.terminate() is called, which means self.runner.monitor_forget will always try to run with the fd of unknown state. The fd could have been forgotten  ( in stream monitor) before it reaches here in cases of hangup and error.